dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/harvest/client/HarvestingClientsPage.java

```diff
@@ -372,39 +353,26 @@ public void createClient(ActionEvent ae) {
         // set default description - they can customize it as they see fit:
         newHarvestingClient.setArchiveDescription(BundleUtil.getStringFromBundle("harvestclients.viewEditDialog.archiveDescription.default.generic"));
 
+        HarvestingClient finalNewHarvestingClient = newHarvestingClient;
+        Try<HarvestingClient> createHarvestingClientOperation = Try.of(() -> harvestingClientsService.createHarvestingClient(finalNewHarvestingClient));
 
-        // will try to save it now:
-
-        try {
-            newHarvestingClient = engineService.submit(new CreateHarvestingClientCommand(dvRequestService.getDataverseRequest(), newHarvestingClient));
-
+        if(createHarvestingClientOperation.isSuccess()) {
```

Comment:
This operation can be in onSuccess() and else in onFailure(). It would be more readable IMO.

Reply:
Since I cannot make assignment `newHarvestingClient = createHarvestingClientOperation.get();` inside lambda I deemed it more consistent to wrap all 'success' an `failure` actions together - hence they are in `if else` statement.